### ISSUE TITLE
Dropdown close after date select

### ISSUE THREAD

**Issue by dl7** [valor-software/ng2-bootstrap#5357]
### Bug description or feature request:

I'm creating a component to filter data in a table. The component is using dropdown. In `dropdownMenu` I can put various types of input e.g. checkboxes, text, datepicker.
Checkboxes and text work very well, but I've problem with datepicker. When I select a date, both datepicker and dropdown close, but I want to dropdown stay open.

It is possible to add `[autoClose]="false"` property, however I would like to keep the auto close behavior after outside click.

![GIF](https://imgur.com/N7CiK4T.gif)

There is any solution to prevent dropdown close after date select?

### Plunker/StackBlitz that reproduces the issue:

StackBlitz: https://stackblitz.com/edit/angular-nusk6m

### Versions of ngx-bootstrap, Angular, and Bootstrap:

ngx-bootstrap: 5.1.2

Angular: 8.2.0

Bootstrap: 4.3.1

**Comment by ludmilanesvitiy** [comment]
@dl7 You can set [autoClose]="false" to the dropdown, like in this example: https://valor-software.com/ngx-bootstrap/#/dropdowns#autoclose

**Comment by dl7** [comment]
@ludmilanesvitiy 
However, as I mentioned, I would like to keep the auto close behavior after outside click. If I set [autoClose]="false", it won't work.

**Comment by faresk93** [comment]
I have the same problem.
Any solution ?

**Comment by sujeetjaiswara** [comment]
Same problem.
Any solution ?

**Comment by PieterjanDeClippel** [comment]
Same here

**Comment by nikolastj** [comment]
Actually I found the solution that worked for me: **bs-datepicker-inline** and implement custom dropdown element on ng-select

**Comment by anuragxcode** [comment]
i was facing the same issue, i fixed through adding class to my ngbDropdown and the button which click i reopen the dropdown.

@HostListener('document:click', ['$event'])
  onClick(event: MouseEvent) {
    if (this.dateFilterButton.nativeElement.contains(event.target) || this.dateFilterSpan.nativeElement.contains(event.target)) {
      this.dropDown.open();
    } else if (this.dropDown && !this.ddEl.nativeElement.contains(event.target) && this.dropDown.isOpen() && !document.querySelector(".cdk-overlay-backdrop")) {
      this.close();
    }
  }
  
  also need to find the appended call while you open the datepicker.
  Ex : !document.querySelector(".cdk-overlay-backdrop")

**Comment by anuragxcode** [comment]
I have solve my problem through below approach.

below is my html structure.

<div class="btn-group dropdown filter-dd" [ngClass]="{'flat' : hideTriggerEl}" #dropDown="ngbDropdown" ngbDropdown
  (openChange)="onToggle($event)" [placement]="placement" [autoClose]="false">
  <button #dateFilterButton class="btn btn-default dropdown-toggle" type="button" ngbDropdownToggle [title]="dd.title">
  </button>
  <div class="dropdown-menu filter-dd-menu date-range" ngbDropdownMenu #ddEl>
     // dropdown code and date-picker
  </div>
 </div>
 
 **// .ts file changes**
 
  @ViewChild('dropDown', { static: true }) dropDown: NgbDropdown;
  @ViewChild('ddEl') ddEl: ElementRef; // Reference to the dropdown menu element
  @ViewChild('dateFilterButton') dateFilterButton: ElementRef;
  @ViewChild('dateFilterSpan') dateFilterSpan: ElementRef;
  
  
 @HostListener('document:click', ['$event'])
  onClick(event: MouseEvent) {
    if (this.dateFilterButton.nativeElement.contains(event.target) || this.dateFilterSpan.nativeElement.contains(event.target)) {
      this.dropDown.open();
    } else if (this.dropDown && !this.ddEl.nativeElement.contains(event.target) && this.dropDown.isOpen() && !document.querySelector(".cdk-overlay-backdrop")) {
      this.close();
    }
  }
  
  
  
  basically need to make reference of the button and the ngbDropdown and close when click outside and check if you click on the button then open the dropdown and also can check for the cdk-overlay that appended for the Datepicker.
